components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddonCollectionProvider.kt

```diff
@@ -64,11 +64,14 @@ class AddonCollectionProvider(
      *
      * @param allowCache whether or not the result may be provided
      * from a previously cached response, defaults to true.
+     * @param readTimeoutInSeconds optional timeout in seconds to use when fetching
+     * available add-ons from a remote endpoint. If not specified [DEFAULT_READ_TIMEOUT_IN_SECONDS]
+     * will be used.
      * @throws IOException if the request failed, or could not be executed due to cancellation,
      * a connectivity problem or a timeout.
      */
     @Throws(IOException::class)
-    override suspend fun getAvailableAddons(allowCache: Boolean): List<Addon> {
+    override suspend fun getAvailableAddons(allowCache: Boolean, readTimeoutInSeconds: Long?): List<Addon> {
```

Comment:
nit:
I'd prefer `readTimeoutInSeconds: Long? = DEFAULT_READ_TIMEOUT_IN_SECONDS` since it makes it obvious at first glance at the function signature what the default is; as-is, you need to read through the method to find out what happens when `readTimeoutInSeconds` is null.

... but I to do this, I think you'd need to declare the default at the interface level, and change it from optional to a required value.

Reply:
... at which point, it's no longer super obvious what's happening when you actually read the implementation (since the default value is one step removed). So, yeah... Nevermind, I guess :)

Reply:
:) Yeah, I came down that same path. This seems to be the best compromise.